test/dynamo/test_functions.py

```diff
@@ -330,6 +332,58 @@ def test_methodcall2(a, b):
     def test_methodcall3(a, b):
         return constant3(a, b=1.0) + b
 
+    def test_is_integer(self):
+        @torch._dynamo.optimize(nopython=True)
+        def forward(t, m):
+            return t if m.is_integer() else 2 * t
+
+        t = torch.tensor([1])
+        assert forward(t, 1.0).item() == 1
+        assert forward(t, 1.5).item() == 2
+
+        assert forward(t, np.float32(1.0)).item() == 1
+        assert forward(t, np.float64(1.5)).item() == 2
```

Comment:
For these see
https://pytorch.org/docs/main/torch.compiler_faq.html#does-numpy-work-with-torch-compile
in particular:
>NumPy scalars: We model them as 0-D arrays. That is, np.float32(3) returns a 0-D array under torch.compile. To avoid a graph break, it is best to use this 0-D array. If this breaks your code, you can workaround this by casting the NumPy scalar to the relevant Python scalar type bool/int/float.

To support that in that case, you might want to have a look at
https://github.com/pytorch/pytorch/blob/e4c51d22c54ae4bf950b7352f8f058ece148a323/torch/_dynamo/variables/tensor.py#L1079

Reply:
Oh, darn, you are ahead of us. Yes, Andrew and I figured this out, and I'm in the middle of doing that.

However, there's another issue, so I'm going to get rid of supporting numpy scalars, and instead file an issue with a case to reproduce it.